### PR TITLE
Update README.md to use `pub add` instead of explicit version

### PR DESCRIPTION
**Pull request by spydon**
This updated the README.md to use `pub add` instead of adding an explicit version to `pubspec.yaml`.

This makes it easier to maintain, since the version isn't explicitly mentioned and it is also less error prone for the user.

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read the [Flutter Style Guide] _recently_, and have followed its advice.
- [x] I signed the [CLA].
- [-] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or there is a reason for not adding tests.

Please add the `release-notes-not-required`, since it shouldn't need an entry there.

### CODE REVIEW

**Review by kenzieschmoll — COMMENTED** [review]
Thanks for this. If you wouldn't mind, could you make the same correction to the devtools_extensions readme?

Also, please add an entry to the respective CHANGELOG files. The entry for devtools_app_shared/CHANGELOG can be made under the 0.1.0 section since that has not been published yet. For devtools_extensions, you can use the 0.0.15-wip section.

**Comment by spydon** [comment]
> Thanks for this. If you wouldn't mind, could you make the same correction to the devtools_extensions readme?
> 
> Also, please add an entry to the respective CHANGELOG files. The entry for devtools_app_shared/CHANGELOG can be made under the 0.1.0 section since that has not been published yet. For devtools_extensions, you can use the 0.0.15-wip section.

Sure, fixed!

On an unrelated note, have you tried [Melos](https://github.com/invertase/melos)? With Melos you can get the changelogs generated automatically from conventional commits, which makes it easier to maintain since you never have to manually add entries to the changelog and no more changelog conflicts when you update a branch for example.
I bet there is a structure that all Flutter repositories follow so that you wouldn't be able to use it, but if you are able to I'd definitely suggest that you play around a bit with it! It made our lives so much easier for maintaining the Flame monorepo. :)

File: packages/devtools_extensions/README.md [hunk 1]
```diff
@@ -164,10 +164,9 @@ some_pkg/  # formerly the repository root of your pub package
 
 2. Add the `devtools_extensions` dependency to your Flutter web app.
 
-    In `pubspec.yaml`, add the following:
-    ```yaml
-    devtools_extensions: ^0.0.14
-    ```
+   ```console
+   $ flutter pub add devtools_app_shared
```
Comment (kenzieschmoll) [inline packages/devtools_extensions/README.md:168]: this needs to be devtools_extensions

**Review by kenzieschmoll — APPROVED** [review]
one correction request, but lgtm

**Comment by kenzieschmoll** [comment]
thanks for the Melos suggestion - filed https://github.com/flutter/devtools/issues/7399 to track.

File: packages/devtools_extensions/README.md (hunk 1 above)
Comment (spydon) [reply packages/devtools_extensions/README.md:168]: Haha oops, copy paste mistake :sweat_smile:

Comment (spydon) [inline packages/devtools_extensions/README.md:168]:
```suggestion
   $ flutter pub add devtools_extensions
```

**Review by kenzieschmoll — APPROVED** [review]

**Comment by kenzieschmoll** [comment]
landing on red since these are just readme updates. Bots failing on a known integration test flake.

**Pull request merged**